### PR TITLE
azure - Remove Azure Functions extensions dlls

### PR DESCRIPTION
**Pull request by logachev**
-Replace actual extension dlls with new `extensionBundle` feature.
-Add missing headers for dedicated plans zip deploy

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/function_package.py [hunk 1]
```diff
@@ -99,16 +97,11 @@ def _add_host_config(self):
                     }
                 }
             }
+        if mode == FUNCTION_EVENT_TRIGGER_MODE:
+            config['extensionBundle'] = {"id": "Microsoft.Azure.Functions.ExtensionBundle",
+                                         "version": "[1.*, 2.0.0)"}
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/function_package.py]: Is it worth pulling this version value out to the `Constants` file?

File: tools/c7n_azure/c7n_azure/function_package.py [hunk 2]
```diff
@@ -228,13 +221,21 @@ def publish(self, deployment_creds):
         # update perms of the package
         self._update_perms_package()
         zip_api_url = '%s/api/zipdeploy?isAsync=true' % deployment_creds.scm_uri
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/function_package.py]: Why are the user credentials required? I am surprised `deployment_creds.scm_uri` is not sufficient here.

File: tools/c7n_azure/tests/test_function_package.py [hunk 3]
```diff
@@ -103,8 +103,20 @@ def test_event_package_files(self, session_mock):
...
         self.assertTrue(FunctionPackageTest._file_exists(files, 'test-azure-package/config.json'))
         self.assertTrue(FunctionPackageTest._file_exists(files, 'host.json'))
-        self.assertTrue(FunctionPackageTest._file_exists(files, 'extensions.csproj'))
-        self.assertTrue(FunctionPackageTest._file_exists(files, 'bin/extensions.dll'))
+
+    def test_add_host_config(self, add_contents_mock):
+        packer = FunctionPackage('test')
+        with patch('c7n.mu.PythonPackageArchive.add_contents') as mock:
+            packer._add_host_config(FUNCTION_EVENT_TRIGGER_MODE)
+            mock.asser_called_once()
```
Comment (aluong) [inline tools/c7n_azure/tests/test_function_package.py]: Is `asser_called_once` a typo and it should be `assert_called_once`?

**Comment by aluong** [comment]
Very happy with PR =)

File: tools/c7n_azure/c7n_azure/function_package.py (hunk 1 above)
Comment (logachev) [reply tools/c7n_azure/c7n_azure/function_package.py]: I think if we pull version from constants, we should pull all other config values as well.. What do you think?

Comment (aluong) [reply tools/c7n_azure/c7n_azure/function_package.py]: I would be fine with that.

File: tools/c7n_azure/tests/test_function_package.py [hunk 4]
```diff
@@ -103,8 +105,40 @@ def test_event_package_files(self, session_mock):
...
+        creds = User(publishing_user_name='user',
+                     publishing_password='password',
+                     scm_uri='https://uri')
+
+        packer.publish(creds)
+
+        post_mock.assert_called_once()
+        status_mock.raise_for_status.assert_called_once()
+
+        print(post_mock.call_args)
```
Comment (aluong) [inline tools/c7n_azure/tests/test_function_package.py]: I don't think you need this print statement.

**Pull request merged**